### ISSUE TITLE
any_executor.hpp does not compile with -fno-rtti under gcc

### ISSUE THREAD

**Issue by edtanous** [chriskohlhoff/asio#533]
I found this behavior when upgrading from boost 1.73.0 to boost 1.74.0.  The newly created any_executor function relies on typeid explicitly, here https://github.com/chriskohlhoff/asio/blob/be7badc31abcc395cf868de6a1e240c2350bdbf2/asio/include/asio/execution/any_executor.hpp#L810

And does not seem to obey the BOOST_NO_TYPEID config flag or the ASIO_NO_TYPEID flag.  The specific error when compiling with gcc-10 is:
```
| ../recipe-sysroot/usr/include/boost/asio/execution/any_executor.hpp: In static member function 'static const std::type_info& boost::asio::execution::detail::any_executor_base::target_type_void()':
| ../recipe-sysroot/usr/include/boost/asio/execution/any_executor.hpp:811:23: error: cannot use 'typeid' with '-fno-rtti'
|   811 |     return typeid(void);
```

I have very little context about what any_executor does, this just showed up in a #include <asio.hpp> include in a project I was trying to upgrade.

**Comment by brunoabinader** [comment]
@edtanous I made a patch fix in #541: could you please verify if that works for you?

**Comment by edtanous** [comment]
Just applied it to my local tree, and seems to resolve the issue.  Thanks for the quick turnaround!

**Comment by jmog1994** [comment]
Hey I just found the same issue trying to import boost 1.74 in a legacy project which uses -fno-rtti I am a bit new here how this patch may be applied to the library it is downloaded...

**Comment by chriskohlhoff** [comment]
This was fixed in asio 1.18.1 / boost 1.75.
